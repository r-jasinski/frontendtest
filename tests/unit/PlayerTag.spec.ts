import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import Player from '../../src/components/PlayerTag.vue'

describe('Player.vue', () => {
  const wrapper = mount(Player, {
    propsData: {
      flag: 'https://i.imgur.com/flag.png',
      image: 'https://i.imgur.com/image.png',
      isPlayerTurn: true,
      name: 'Guest9241962816',
      time: '4:58'
    }
  })
  it('should render image', () => {
    expect(wrapper.find('img').attributes('src')).toBe(
      'https://i.imgur.com/image.png'
    )
  })
  it('should render name', () => {
    expect(wrapper.find('span').text()).toBe('Guest9241962816')
  })
  it('should render flag', () => {
    expect(wrapper.find('.user-tagline img').attributes('src')).toBe(
      'https://i.imgur.com/flag.png'
    )
  })
  it('should render network status', () => {
    expect(wrapper.findAll('li')).toHaveLength(4)
  })
  it('should render time', () => {
    expect(wrapper.find('.clock').text()).toBe('4:58')
  })
  it('should have isPlayerTurn property set to true', () => {
    expect(wrapper.props('isPlayerTurn')).toBe(true)
  })
  it('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
