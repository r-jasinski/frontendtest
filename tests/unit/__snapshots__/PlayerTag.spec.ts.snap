// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Player.vue > should match snapshot 1`] = `
<div
  class="player"
  data-v-a6c9493b=""
>
  <div
    class="tagline"
    data-v-a6c9493b=""
  >
    <img
      alt="Player image"
      data-v-a6c9493b=""
      src="https://i.imgur.com/image.png"
    />
    <div
      class="user-tagline"
      data-v-a6c9493b=""
    >
      <span
        data-v-a6c9493b=""
      >
        Guest9241962816
      </span>
      <img
        alt="Player flag"
        data-v-a6c9493b=""
        src="https://i.imgur.com/flag.png"
      />
      <ul
        data-v-a6c9493b=""
      >
        <li
          data-v-a6c9493b=""
        />
        <li
          data-v-a6c9493b=""
        />
        <li
          data-v-a6c9493b=""
        />
        <li
          data-v-a6c9493b=""
        />
      </ul>
    </div>
  </div>
  <div
    class="clock"
    data-v-a6c9493b=""
  >
    </src/assets/icons/clock.svg?component
      data-v-a6c9493b=""
    />
    <span
      data-v-a6c9493b=""
    >
      4:58
    </span>
  </div>
</div>
`;
